tests/behaviour/typeql/reasoner/steps.rs

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+use cucumber::{gherkin::Step, given, then, when};
+use typedb_client::TransactionType;
+
+use crate::{
+    behaviour::{
+        connection::{
+            database::steps::connection_create_database,
+            session::steps::{
+                connection_close_all_sessions, connection_open_data_session_for_database,
+                connection_open_schema_session_for_database,
+            },
+            transaction::steps::{session_opens_transaction_of_type, transaction_commits},
+        },
+        parameter::TransactionTypeParam,
+        typeql::language::steps::{answer_size, get_answers_typeql_match, typeql_define, typeql_insert},
+        Context,
+    },
+    generic_step_impl,
+};
+
+generic_step_impl! {
+    #[step(expr = "reasoning schema")]
+    async fn reasoning_schema(context: &mut Context, step: &Step) {
+        if context.databases.all().await.unwrap().is_empty() {
```

Comment:
Shouldn't this be `!context.databases.contains()`?

Reply:
Yeah, it came from usual step "connection does not have any database", but we really need to check only one database here.